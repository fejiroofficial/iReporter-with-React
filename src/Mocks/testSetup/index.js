/* eslint-disable import/no-extraneous-dependencies */
import '@babel/polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import configMockStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';


export const mockState = {
  user: {
    token: '',
    isLoggedIn: false,
    loginStatus: false,
    message: '',
    isLoading: false,
    profile: {
      firstname: 'John',
      username: 'johnny',
      image: 'image.jpg',
      errorData: [],
    },
    resetPassword: {
      loading: false,
      message: '',
    }
  },
  bookmark: {
    loading: false,
    articles: [],
    message: '',
  },
  article: {
    newComments: {
      body: 'Hello world',
      id: 1,
      authorName: 'John',
      authorImage: 'image.jpg',
      date: 'July 4th 2018',
    },
    postingComment: false,
    commentMessage: 'Done',
    gettingComments: false,
    oldComments: [],
    hasMoreComments: false,
    commentPage: {
      current: 1,
      currentCount: 1,
      totalCount: 1,
    },
  },
  createArticle: {
    loading: false,
    message: false,
  },
  articlesReducer: {
    articles: []
  },
  articlesCategoryReducer: []
};


export const createMockStore = () => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  return mockStore;
};

const setup = (component, initailState = mockState) => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  const connectedWrapper = mount(
    <Provider store={mockStore(initailState)}>
      <BrowserRouter>
        {component}
      </BrowserRouter>
    </Provider>
  );
  return connectedWrapper;
};

export default setup;
