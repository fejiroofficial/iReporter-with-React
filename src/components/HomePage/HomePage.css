.InfoBox {
    display: flex;
    flex-wrap: wrap;
    width: fit-content;
}

.InfoBoxMini {
    min-height: 250px;
    width: 270px;
    margin-left: 26px;
    margin-bottom: 60px;
    border-style: solid;
    border-width: thin;
    border-radius: 7px;
    border-color: lightgrey;
}

.Box {
    width: inherit;
    height: fit-content;
    text-align: -webkit-center;
    text-align: -moz-center;
}

.InfoBoxMiniImg{
    height: 154px;
    width: inherit;
    background-image: url('../.././assets/img/megaphone.jpg');
    background-size: 64%;
    background-position: center;
    background-repeat: no-repeat;
}

.InfoBoxMiniImgTwo{
    height: 154px;
    width: inherit;
    background-image: url('../.././assets/img/investigate.jpg');
    background-size: 50%;
    background-position: center;
    background-repeat: no-repeat;
}

.InfoBoxMiniImgThree{
  height: 154px;
  width: inherit;
  background-image: url('../.././assets/img/newspaper.jpg');
  background-size: 69%;
  background-position: center;
  background-repeat: no-repeat;
}

.InfoBoxMiniTitle{
    height: 27px;
    width: inherit;
    text-align: center;
    font-weight: bolder;
}

.InfoBoxMiniText{
  min-height: 48px;
  width: inherit;
  text-align: center;
  margin-top: -12px;
  font-size: smaller;
}

.HomePageMapDiv {
  height: 400px;
  width: inherit;
  background-image: url('../.././assets/img/africaMap3.jpg');
  background-size: contain;
  background-position: center;
  background-repeat: no-repeat;
}

.IncidentContainer {
  display: flex;
  flex-wrap: wrap;
}

.CenterItems {
  height: fit-content;
  width: fit-content;
}

.IncidentMini {
  min-height: 250px;
  width: 270px;
  margin-bottom: 60px;
  margin-right: 40px;
  border-radius: 5px;
}

.PartnersMini {
  height: 150px;
  margin-left: 84px;
  width: 150px;
}

.IncidentMiniOne {
  background-image: url('../.././assets/img/Corruption.jpg');
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
}

.IncidentMiniTwo {
  background-image: url('../.././assets/img/Corruption2.jpg');
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
}

.IncidentMiniThree {
  background-image: url('../.././assets/img/Corruption3.jpg');
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
}

.IncidentMiniFour {
  background-image: url('../.././assets/img/yahoo.jpg');
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
}

.IncidentMiniFive {
  background-image: url('../.././assets/img/bridge.jpg');
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
}

.IncidentMiniSix {
  background-image: url('../.././assets/img/FCT-Road.jpg');
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
}

.IncidentMiniSeven {
  background-image: url('../.././assets/img/marine-litter.jpg');
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
}

.IncidentMiniEight {
  background-image: url('../.././assets/img/Oil-spill.png');
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
}

.PartnersMiniOne {
  background-image: url('../.././assets/img/andela.jpg');
  background-size: 69%;
  background-position: center center;
  background-repeat: no-repeat;
}

.PartnersMiniTwo {
  background-image: url('../.././assets/img/thinkAfrica.jpg');
  background-size: contain;
  background-position: center center;
  background-repeat: no-repeat;
}

.PartnersMiniThree {
  background-image: url('../.././assets/img/now.jpg');
  background-size: contain;
  background-position: center center;
  background-repeat: no-repeat;
}

.PartnersMiniFour {
  background-image: url('../.././assets/img/unicef.jpg');
  background-size: contain;
  background-position: center center;
  background-repeat: no-repeat;
}

@media (min-width: 1024px) {
  .HomePageMapDiv {
    margin-bottom: 41px;
    height: 697px;
  }
}
