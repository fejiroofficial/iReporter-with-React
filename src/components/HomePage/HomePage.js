import React from 'react';
import Aux from '../../hoc/Aux';

const HomePage = () => (
  <Aux>
    <div>Hero-Banner</div>
    <div>OUR WORK IS IMPORTANT</div>
    <div>Map-photo</div>
    <div>Sample photos redflag</div>
    <div>Sample photos interventions</div>
    <div>Our partners</div>
  </Aux>
)

export default HomePage;