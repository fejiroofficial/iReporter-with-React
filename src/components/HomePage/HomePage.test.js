import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { createMockStore } from '../../Mocks/testSetup';
import HomePage from './HomePage';

createMockStore();

describe('<HomePage>', () => {
  it('should render without crashing', () => {
    const container = (
      <HomePage>
        <p>Hello</p>
      </HomePage>
    );
    const wrapper = shallow(container);
    expect(wrapper).toMatchSnapshot();
  });
});
