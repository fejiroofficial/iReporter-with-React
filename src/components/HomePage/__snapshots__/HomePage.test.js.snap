// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage> should render without crashing 1`] = `
<aux>
  <Connect(heroBanner) />
  <MiniHolder
    header="OUR WORK IS IMPORTANT"
  >
    <div>
      <div>
        <div>
          <div />
          <div>
            REPORT
          </div>
          <div>
            <p>
              We provide a platform for reporting corruption and intervention cases to us.
            </p>
          </div>
        </div>
        <div>
          <div />
          <div>
            INVESTIGATE
          </div>
          <div>
            <p>
              We investigate selected reports of alleged red-flag and intervention cases that have an impact on society.
            </p>
          </div>
        </div>
        <div>
          <div />
          <div>
            EXPOSE
          </div>
          <div>
            <p>
              We work with mainstream media and security agencies to ensure corruption is fully exposed.
            </p>
          </div>
        </div>
      </div>
    </div>
  </MiniHolder>
  <MiniHolder
    header="WE COVER AFRICA"
  >
    <div />
  </MiniHolder>
  <MiniHolder
    header="RED-FLAGS"
  >
    <div>
      <div
        className="undefined undefined"
      />
      <div
        className="undefined undefined"
      />
      <div
        className="undefined undefined"
      />
      <div
        className="undefined undefined"
      />
    </div>
  </MiniHolder>
  <MiniHolder
    header="INTERVENTIONS"
  >
    <div>
      <div
        className="undefined undefined"
      />
      <div
        className="undefined undefined"
      />
      <div
        className="undefined undefined"
      />
      <div
        className="undefined undefined"
      />
    </div>
  </MiniHolder>
  <MiniHolder
    header="OUR PARTNERS"
  >
    <div>
      <div
        className="undefined undefined"
      />
      <div
        className="undefined undefined"
      />
      <div
        className="undefined undefined"
      />
      <div
        className="undefined undefined"
      />
    </div>
  </MiniHolder>
</aux>
`;
